### DISCUSSION TITLE
a/python-project-manager-uv

### DISCUSSION THREAD

**Discussion by giscus** [sincerefly/giscus-data#18] [Announcements]
# a/python-project-manager-uv

https://blog.yasking.org/a/python-project-manager-uv.html

**Comment by pxladx** [comment]
大神，请问uv 使用的时候如何像pip.conf这样设置源。因为使用uv管理项目的时候安装库都很慢，我想使用国内的镜像源。

**Reply by sincerefly** [reply]
单次安装的时候可以使用 `--index-url` 参数，例如 `uv pip install requests --index-url https://mirrors.tuna.tsinghua.edu.cn/pypi/web/simple`

也可以设置环境变量：UV_INDEX_URL=https://mirrors.tuna.tsinghua.edu.cn/pypi/web/simple

目前看官方并没有提供持久化的换源方案，这个 Issues （[#1404](https://github.com/astral-sh/uv/issues/1404)）有一些讨论

**Reply by pxladx** [reply]
非常感谢，这个讨论我之前也看到了。

我不是专业程序员，有个小白问题，想请问一下：
是不是通过uv创建项目以后，在pyproject.toml这个文件里面添加UV_INDEX_URL，就可以算作添加镜像源了？就像下面这样：
[tool.uv]
index-url = "https://mirrors.tuna.tsinghua.edu.cn/pypi/web/simple"

**Reply by sincerefly** [reply]
@pxladx 我没具体测试，不过大概率不行，参考这两个 Issues（[#8166](https://github.com/astral-sh/uv/issues/8166) [#8186](https://github.com/astral-sh/uv/issues/8186)）有讨论到你粘贴的写法，看着还是建议状态，没有实现这个功能

你自己测试的话，可以使用 -v 参数，从输出详情可以看到包是从哪里下载的，以判断 `index-url` 是否生效

```bash
$ uv pip install requests -v
```

感觉 uv 官方对于从多个位置（环境变量、全局配置、用户配置、项目配置...）获取配置的支持很谨慎，目前还没考虑/设计好 🤔

**Reply by pxladx** [reply]
好的，非常感谢，我试一试。
